src/blockassembler.cpp

```diff
@@ -226,7 +226,7 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
         if (chainActive.Tip() != pindexPrev) return nullptr; // new block came in, move on
 
         CValidationState state;
-        if (!TestBlockValidity(state, *pblock, pindexPrev, false, false)) {
+        if (!TestBlockValidity(state, *pblock, pindexPrev, false, false, false)) {
```

Comment:
The reason behind every check flag false is to not do double validations and slowdown the block generation process when the same wallet crafted the block and knows what is inside it + followed every consensus rule creating it.

If the signature fails for any reason, it will fail in the `SignBlock` function call that is few lines above of this line. The only reason to have`TestBlockValidity::CheckBlock()` call failing is because (1) there is a problem in the block generation code that is not following the consensus rules or (2) the user modified the miner sources in some way (which we don't care).

Reply:
> If the signature fails for any reason, it will fail in the SignBlock function call that is few lines above of this line.

Assuming no bugs (or hw failures) in the signature process.
`SignBlock` only calls `CKey::Sign`, there is no verification.
`CKey::Sign` calls `secp256k1_ecdsa_sign` and `secp256k1_ecdsa_signature_serialize_der`. It can return false only if the key is not found, or not valid.
In `CPubKey::Verify`, instead, the signature is verified with `secp256k1_ec_pubkey_parse` and `secp256k1_ecdsa_verify`.

Of course, adding the check in `TestBlockValidity` would make it a bit slower... but maybe it's worth.
I'm fine anyway (that's why have already acked).